assets/app/scripts/services/applicationGenerator.js

```diff
@@ -0,0 +1,270 @@
+"use strict";
+
+angular.module("openshiftConsole")
+
+  .service("ApplicationGenerator", function(){
+    var osApiVersion = "v1beta1";
+    var k8sApiVersion = "v1beta3";
+    
+    var service = {};
+    var scope = service;
+    
+    scope._generateSecret_ = function(){
+        //http://stackoverflow.com/questions/105034/create-guid-uuid-in-javascript
+        function s4() {
+          return Math.floor((1 + Math.random()) * 0x10000)
+              .toString(16)
+              .substring(1);
+          }
+        return s4()+s4()+s4()+s4();
+      };
+
+    /**
+    * Find the 'first' port of exposed ports.
+    * @param            ports  list of ports (e.g {containerPort: 80, protocol: "tcp"})
+    * @return {integer} The port/protocol pair of the lowest conainer port
+    */
+    scope._getFirstPort_ = function(ports){
+      var first = "None";
```

Comment:
The value of "None" is used when specifying a headless service.

On 04/08/2015 03:18 PM, Jessica Forrester wrote:
> In assets/app/scripts/services/applicationGenerator.js
> <https://github.com/openshift/origin/pull/1221#discussion_r28003419>:
>
>> +        //http://stackoverflow.com/questions/105034/create-guid-uuid-in-javascript
>> +        function s4() {
>> +          return Math.floor((1 + Math.random()) * 0x10000)
>> +              .toString(16)
>> +              .substring(1);
>> +          }
>> +        return s4()+s4()+s4()+s4();
>> +      };
>> +
>> +    /**
>> +    * Find the 'first' port of exposed ports.
>> +    * @param            ports  list of ports (e.g {containerPort: 80, protocol: "tcp"})
>> +    * @return {integer} The port/protocol pair of the lowest conainer port
>> +    */
>> +    scope._getFirstPort_ = function(ports){
>> +      var first = "None";
>
> are we actually passing "None" back to the API somewhere? otherwise it
> seems like this should return null when there is not a port
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/1221/files#r28003419>.
>

-- 
Jeff Cantrill
Senior Software Engineer, Red Hat Engineering
Red Hat
Office: 703-748-4420 | 866-546-8970 ext. 8162420
jcantril@redhat.com
http://www.redhat.com